text/3724-theupdateframework.md

```diff
@@ -0,0 +1,341 @@
+- **Threat**: Misuse of the `Crates Role` to sign a malicious crate index file (whether by an entity with access to that key or through compromise)
+  **Addressed**: The snapshot/timestamp roles handle revoking the malicious signature, and this is otherwise handled identically to compromise of the `Crates Role`.
+- **Threat**: Compromise of the `tuf-crates Root Role`
+  **Addressed**: This scenario requires the compromise of `N` crates.io team members. Handled via the revocation and rotation via the Rust root key quorum.
+- **Threat**: Compromise or loss of 1-4 keys in the Rust root quorum
+  **Addressed**: Handled via 5+ members of the root quorum signing a new root with those keys rotated out and new keys rotated in.
+- **Threat**: Compromise or loss of 5+ keys in the Rust root quorum
+  **Addressed**: Manual replacement and transition of Rust root quorum, featuring loud announcement from Rust project, signed by remaining quorum members, corroborated by news of whatever disaster led to this event.
+- **Threat**: Compromise or MITM of the `crates.io` TLS certificate used for HTTPS
+  **Addressed**: All signatures are independent of TLS, so the compromised/MITMed connection cannot tamper with the indexes (see threats in previous section). The `Crates Role` provided by crates.io chains up to the Rust root role, so the compromised/MITMed connection cannot present a different `Crates Role`.
+- **Threat**: Presentation of an old/revoked/expired `tuf-crates` signature
+  **Addressed**: TUF implements quorum history in a way that cargo will identify that it is not the latest quorum. The `Timestamp Role` will also mitigate the usage of an old signature by limiting the attack period to N-hours, as configured by crates.io.
+
+## Artifact Signing (`tuf-crates`)
+
+Instead of signing the index entries and transitively inferring security via those signatures, we could alternatively or additionally have the TUF repository include direct signatures of the artifacts. However, omitting signing of the index entries would allow various threats (see "Threat Model"), and signing the individual crate files does not provide added security above and beyond signing index files that include cryptographic hashes of the individual crate files.
+
+# Prior art
+[prior-art]: #prior-art
+
+Previous RFC, not based on TUF; closed in favor of this one:
+
+- [RFC #3579: Public Key Infrastructure for Rust Project](https://github.com/rust-lang/rfcs/pull/3579)
+
+Relevant past RFCs:
+
+- [RFC #3403: sigstore and cargo/crates.io](https://github.com/rust-lang/rfcs/pull/3403)
+- [TUF for crates.io discussion](https://github.com/withoutboats/rfcs/pull/7)
+- [Trusted Publishing support for crates.io](https://github.com/rust-lang/rfcs/pull/3691)
+
+Other posts on TUF usage:
+- [Python PEP #458: Secure PyPI downloads with signed repository metadata](https://peps.python.org/pep-0458/)
```

Comment:
I mentioned this in the comment about about hardware-backed quorums, but to highlight it here as well: PEP 458 was never fully rolled out on PyPI, in part because we had trouble operationalizing the signing model. So that doc specifies an _architecture_ for TUF on PyPI, but it's worth nothing that the _implementation_ itself isn't in place.